### PR TITLE
Add Files to support compilation with Visual Studio 2017 Community Edition (Chibios)

### PR DESCRIPTION
**Pull request by TerryFogg**
Added
1. CMakeSettings.Json
  a. Example for STM32F769IDiscovery board - Compiling
  b. Example ESP32 - needs additional checking, which variable and where to find FREERTOS

2. Change .gitignore 
 a. Keep build folder which is required to exist for successful operation with Visual STudio
b. Keep launch.cs.json - example launch for debugging

3. New file launch.vs.json - example for STM32F769IDiscovery board

4. change to FindBuildUtils.cmake to ignore build for Visual Studio due to bug in VS

5. CopyBuildOutput.cmd - File to generate outputs ( workaround due to VS bug) must be run manually

7. Other changes due to formatting should be ignore, due to auto formatting by Visual Studio  (style)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,8 +1,9 @@
 # ignore all ZIP files
 *.zip
 
-# ignore build folder
-/build/
+# ignore build folder contents but not the folder
```
Comment (josesimoes) [inline .gitignore]: please fix the merge error that is showing

File: VisualStudioDevelopment/Visual Studio nf-interpreter development.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
+[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0) [![#yourfirstpr](https://img.shields.io/badge/first--timers--only-friendly-blue.svg)](https://github.com/nanoframework/Home/blob/master/CONTRIBUTING.md)
```
Comment (josesimoes) [inline VisualStudioDevelopment/Visual Studio nf-interpreter development.md]: please remove the badges and logo from this, make it start at the tite bellow "Developing firmware..."

File: src/CLR/Include/nanoCLR_Runtime.h [hunk 3]
```diff
@@ -37,7 +37,7 @@ extern const CLR_RADIAN c_CLR_radians[];
 typedef std::set< std::wstring >             CLR_RT_StringSet;
 typedef CLR_RT_StringSet::iterator           CLR_RT_StringSetIter;
 
-typedef std::map< std::string,int >          CLR_RT_StringMap;
+typedef std::map< std::string, int >          CLR_RT_StringMap;
```
Comment (josesimoes) [inline src/CLR/Include/nanoCLR_Runtime.h]: can't have this along with the VS build changes, understand that this is a formatting thing from V. 
please revert the file and commit again

File: src/HAL/Include/nanoHAL_Time.h [hunk 4]
```diff
@@ -98,10 +98,10 @@ HRESULT HAL_Time_AccDaysInMonth(signed int year, signed int month, signed int* d
 uint64_t HAL_Time_ConvertFromSystemTime(const SYSTEMTIME* systemTime);
 
 /// APIs to convert between types
-bool          HAL_Time_TimeSpanToStringEx( const int64_t& ticks, char*& buf, size_t& len );
+bool          HAL_Time_TimeSpanToStringEx(const int64_t& ticks, char*& buf, size_t& len);
```
Comment (josesimoes) [inline src/HAL/Include/nanoHAL_Time.h]: can't have this along with the VS build changes, understand that this is a formatting thing from V. 
please revert the file and commit again

File: src/HAL/nanoHAL_Time.cpp [hunk 5]
```diff
@@ -26,7 +26,7 @@ const int DaysInMonth[12] = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
 
 uint64_t  HAL_Time_CurrentTime()
 { 
-    return HAL_Time_SysTicksToTime( HAL_Time_CurrentSysTicks() ); 
+    return HAL_Time_SysTicksToTime( HAL_Time_CurrentSysTicks() );
```
Comment (josesimoes) [inline src/HAL/nanoHAL_Time.cpp]: can't have this along with the VS build changes, understand that this is a formatting thing from V. 
please revert the file and commit again

File: src/PAL/Include/nanoPAL_Events.h [hunk 6]
```diff
@@ -21,6 +21,7 @@
 #define EVENT_WIFI        50
 #define EVENT_CAN         60
 
+
```
Comment (josesimoes) [inline src/PAL/Include/nanoPAL_Events.h]: no need to touch this here

File: src/CLR/Include/nanoCLR_Runtime.h [hunk 7]
```diff
@@ -1262,6 +1262,14 @@ struct CLR_RT_WellKnownTypes
 
     CLR_RT_TypeDef_Index m_I2cTransferResult;
 
+#if defined(NANOCLR_USE_GRAPHICS)
```
Comment (josesimoes) [inline src/CLR/Include/nanoCLR_Runtime.h]: this doesn't belong in this PR...

File: src/HAL/Include/nanoHAL_Time.h [hunk 8]
```diff
@@ -104,4 +104,7 @@ const char*   HAL_Time_CurrentDateTimeToString();
 
 uint64_t  CPU_MillisecondsToTicks       ( uint64_t ticks  );
 
+
+uint64_t Time_GetMachineTime();
```
Comment (josesimoes) [inline src/HAL/Include/nanoHAL_Time.h]: this doesn't belong in this PR...

File: src/HAL/nanoHAL_Time.cpp [hunk 9]
```diff
@@ -6,8 +6,8 @@
 
 #include <nanoHAL_Time.h>
 
-const int CummulativeDaysForMonth[13] = {0, 31, 59, 90, 120, 151, 181, 212, 243, 273, 304, 334, 365};
-const int DaysInMonth[12] = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
+const int CummulativeDaysForMonth[13] = { 0, 31, 59, 90, 120, 151, 181, 212, 243, 273, 304, 334, 365 };
```
Comment (josesimoes) [inline src/HAL/nanoHAL_Time.cpp]: don't need to touch this...

File: src/PAL/Include/nanoPAL_Events.h [hunk 10]
```diff
@@ -14,13 +14,19 @@
 ////////////////////////////////////////////////////////////////////////////////////////////
 
 #define EVENT_UNKNOWN     0
+#define EVENT_TOUCH       1
```
Comment (josesimoes) [inline src/PAL/Include/nanoPAL_Events.h]: these changes don't belong in this PR...

**Comment by TerryFogg** [comment]
This should be okay now.
Only 8 files, and a phenomenal amount of work just to get it to work.
Terry
